test/compilable/imports/a313templatemixin1.d

```diff
@@ -1,3 +1,4 @@
+module imports.a313templatemixin1;
```

Comment:
Why is this necessary? We should continue to work if the module declaration is not present.

Reply:
It's one way to go.  You could also add the `imports` directory to the "import path list", i.e.
```
dmd -Icompilable/imports ...
```
then no module declaration is required.

Reply:
@marler8997 can't we just penalize people who enter wrong module information? It seems this PR does that but also adds another restriction, which is more difficult to justify.

Reply:
Sorry, I'm not sure exactly what you're trying to say.  There's 2 cases listed in the original PR message that have been deprecated, and 1 that has been declared to be an error.  Do those use cases categorize the cases you are thinking of?  If so, which ones do you think should not be deprecated/made errors?

Reply:
@marler8997 yah let me discuss exactly on the cases you present in the changelog. Thanks for the neat organization.